instrumentation/opentelemetry-instrumentation-aiokafka/tests/test_instrumentation.py

```diff
@@ -11,15 +11,74 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from unittest import TestCase
 
-from aiokafka import AIOKafkaConsumer, AIOKafkaProducer
+import uuid
+from typing import Any, List, Sequence, Tuple
+from unittest import IsolatedAsyncioTestCase, mock
+
+from aiokafka import (
+    AIOKafkaConsumer,
+    AIOKafkaProducer,
+    ConsumerRecord,
+    TopicPartition,
+)
 from wrapt import BoundFunctionWrapper
 
+from opentelemetry import baggage, context
 from opentelemetry.instrumentation.aiokafka import AIOKafkaInstrumentor
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.semconv._incubating.attributes import messaging_attributes
+from opentelemetry.semconv.attributes import server_attributes
+from opentelemetry.test.test_base import TestBase
+from opentelemetry.trace import SpanKind, format_trace_id, set_span_in_context
+
+
+class TestAIOKafka(TestBase, IsolatedAsyncioTestCase):
+    @staticmethod
+    def consumer_record_factory(
+        number: int, headers: Tuple[Tuple[str, bytes], ...]
+    ) -> ConsumerRecord:
+        return ConsumerRecord(
+            f"topic_{number}",
+            number,
+            number,
+            number,
+            number,
+            f"key_{number}".encode(),
+            f"value_{number}".encode(),
+            None,
+            number,
+            number,
+            headers=headers,
+        )
+
+    @staticmethod
+    async def consumer_factory(**consumer_kwargs: Any) -> AIOKafkaConsumer:
+        consumer = AIOKafkaConsumer(**consumer_kwargs)
+
+        consumer._client.bootstrap = mock.AsyncMock()
+        consumer._client._wait_on_metadata = mock.AsyncMock()
+
+        await consumer.start()
+
+        consumer._fetcher.next_record = mock.AsyncMock()
 
+        return consumer
+
+    @staticmethod
+    async def producer_factory() -> AIOKafkaProducer:
+        producer = AIOKafkaProducer(api_version="1.0")
+
+        producer.client._wait_on_metadata = mock.AsyncMock()
+        producer.client.bootstrap = mock.AsyncMock()
+        producer._message_accumulator.add_message = mock.AsyncMock()
+        producer._sender.start = mock.AsyncMock()
+        producer._partition = mock.Mock(return_value=1)
+
+        await producer.start()
+
+        return producer
 
-class TestAIOKafka(TestCase):
     def test_instrument_api(self) -> None:
```

Comment:
I'm not sure you can keep async and not async tests on the same class.

Reply:
Hm. I've never encountered this, but I divided it into two test classes.